### DISCUSSION TITLE
Not able to setup Private NPM with Verdaccio with AWS ELB

### DISCUSSION THREAD

**Discussion by ranuaggarwal1** [verdaccio/verdaccio#3092] [Q&A] [answered]
I am using following Dockerfile and trying to set up private npm with verdaccio with AWS elasticbeanstalk
--------------------------------
FROM verdaccio/verdaccio

USER root

ENV NODE_ENV=production

RUN npm i && npm install verdaccio-s3-storage

COPY config.yaml /verdaccio/conf

COPY htpasswd /verdaccio/conf/htpasswd

USER verdaccio

EXPOSE 4873
-----------------------

I am getting following error
/var/log/eb-docker/containers/eb-current-app/unexpected-quit.log
----------------------------------------
stdout: {"level":40,"time":1648259324263,"pid":8,"hostname":"496ae7d5e3ae","file":"/verdaccio/conf/config.yaml","msg":"config file  - @{file}"}
{"level":50,"time":1648259324278,"pid":8,"hostname":"496ae7d5e3ae","content":"aws-s3-storage","prefix":"verdaccio","msg":"plugin not found. try npm install @{prefix}-@{content}"}
{"level":30,"time":1648259324286,"pid":8,"hostname":"496ae7d5e3ae","msg":"unhandledRejection caught"}

**Comment by juanpicado** [comment]
The stable version of docker verdaccio uses yarn2, not npm, so installing with npm yarn 2 won´t find the plugin, you would need to install with yarn in a docker image, there is some examples around issues area you might find, I do no thave some on my hands now but if I find it i will post it here, but i would recommend you find them.

**Reply by ranuaggarwal1** [reply]
Thank you for the response. I will try it.

**Comment by edclement** [comment, marked as answer]
What version of Verdaccio are you using? You may be able to apply some of the installation instructions provided for the [verdaccio-prometheus-middleware](https://github.com/xlts-dev/verdaccio-prometheus-middleware#installation) plugin.

**Reply by ranuaggarwal1** [reply]
@edclement Thanks a lot for suggesting. I tried this solution and it worked (Referred link https://github.com/verdaccio/verdaccio/issues/1677) when i launched the URL created by elastic beanstalk. But then i had setup an alias url http://npm.domainA.com/. I am getting content security policy issue.
Not sure why this is happening with the alias URL when the original url is working fine.

**Reply by edclement** [reply]
If you look at the network tab in dev tools, where are all the scripts and other content loading from? Is it loading from the alias or the DNS for Elastic Beanstalk? You may want to take a look at this if you haven't yet: https://verdaccio.org/docs/configuration/#url-prefix

**Reply by ranuaggarwal1** [reply]
It is loading from ebs url
![2022-03-28_11-48-17](https://user-images.githubusercontent.com/69798683/160400640-2d13c078-74ab-4d05-b851-7ed027b43315.png)

I will check the link. Thank you for sharing

**Reply by edclement** [reply]
And to clarify my previous comment, I think the config you would be interested in is the `VERDACCIO_PUBLIC_URL` env variable.

**Reply by juanpicado** [reply]
I think that's the solution as far I see. 🙌
